ambassador/ambassador/envoy/v2/v2listener.py

```diff
@@ -500,34 +500,41 @@ def __init__(self, config: 'V2Config', listener: IRListener) -> None:
 
             # Use sane access log spec in JSON
             if(config.ir.ambassador_module.envoy_log_type.lower() == "json") :
+                json_format = {
+                    'start_time': '%START_TIME%',
+                    'method': '%REQ(:METHOD)%',
+                    'path': '%REQ(X-ENVOY-ORIGINAL-PATH?:PATH)%',
+                    'protocol': '%PROTOCOL%',
+                    'response_code': '%RESPONSE_CODE%',
+                    'response_flags': '%RESPONSE_FLAGS%',
+                    'bytes_received': '%BYTES_RECEIVED%',
+                    'bytes_sent': '%BYTES_SENT%',
+                    'duration': '%DURATION%',
+                    'upstream_service_time': '%RESP(X-ENVOY-UPSTREAM-SERVICE-TIME)%',
+                    'x_forwarded_for': '%REQ(X-FORWARDED-FOR)%',
+                    'user_agent': '%REQ(USER-AGENT)%',
+                    'request_id': '%REQ(X-REQUEST-ID)%',
+                    'authority': '%REQ(:AUTHORITY)%',
+                    'upstream_host': '%UPSTREAM_HOST%',
+                    'upstream_cluster': '%UPSTREAM_CLUSTER%',
+                    'upstream_local_address': '%UPSTREAM_LOCAL_ADDRESS%',
+                    'downstream_local_address': '%DOWNSTREAM_LOCAL_ADDRESS%',
+                    'downstream_remote_address': '%DOWNSTREAM_REMOTE_ADDRESS%',
+                    'requested_server_name': '%REQUESTED_SERVER_NAME%',
+                    'istio_policy_status': '%DYNAMIC_METADATA(istio.mixer:status)%',
+                    'upstream_transport_failure_reason': '%UPSTREAM_TRANSPORT_FAILURE_REASON%'
+                }
+
+                tracing_config = config.ir.tracing
+                if tracing_config and tracing_config.driver == 'envoy.tracers.datadog':
```

Comment:
This is so that DataDog APM and logs get correlated together: https://docs.datadoghq.com/tracing/advanced/connect_logs_and_traces/

Until we move to a generic way to define the log json format this is a little hacky but is required to have the APM funcitonality 100% configured, open for other options.